core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -50,6 +50,7 @@ private[spark] class TaskDescription(
     val executorId: String,
     val name: String,
     val index: Int,    // Index within this task's TaskSet
+    val partitionId: Int,
```

Comment:
I'm wondering if we need this. For barrier stage we always retry the entire stage, so `index` must be equal to partitionId.

Reply:
Actually you are right on this, but that makes the logic a little more difficult to understand, so I prefer to use `partitionId` explicitly.